### DISCUSSION TITLE
Adding Group SMS Possible?

### DISCUSSION THREAD

**Discussion by LazyG** [0perationPrivacy/VoIP#109] [Ideas]
I just got this setup following the steps MB outlined.  I love what I see so far, thank you for this!

I was curious if it would be possible to enable a group SMS text option, meaning that a sms messages sent to multiple numbers would be grouped into a single conversation.  Currently, if I try to add 2+ numbers to a message, it creates 2 separate chat histories.

If there is a more appropriate place to make this request please let me know and I will happily move it there.

Thanks again!

**Comment by 0perationPrivacy** [comment]
Unfortunately group sms's are not possible because Twillio and Telnyx don't allow it. They will simply send it as individual texts. It's a VoIP carrier limitation.

**Comment by jvoiptel** [comment]
I was wondering the same thing. It would be nice to support group messaging. I do see that Twilio now supports it. Also, Bandwidth.com supports group messaging as well. 

https://www.twilio.com/docs/conversations/group-texting
https://www.bandwidth.com/messaging/group-messaging-api/

Problem I see is that there is no real standard for group messaging over SMS.  Each provider appears to be implementing their own flavor.

**Reply by 0perationPrivacy** [reply]
I will have a look at that API in the next few days and test it out. Thanks!

**Reply by 0perationPrivacy** [reply]
the Twilio link went dead ☠️

**Reply by LazyG** [reply]
You are correct, I can confirm.  Oddly, they still have their blog post up about the functionality...https://www.twilio.com/blog/group-sms-twilio-java

**Reply by rickoooooo** [reply]
I'm very interested in this project, but group MMS is a requirement for my use case. I can confirm that Twilio does support group MMS using their Conversations API. I have it functioning in my own project (an XMPP to Twilio MMS bridge). However, I believe the API only supports up to four simultaneous MMS users per conversation by default. I think there is a way to request a limit increase up to 10 MMS users though. I can't recall where I read that when I originally was looking into this a year or two ago. Twilio changes things all the time so it may not be accurate any more.

If I recall correctly, you need to first create a "conversation" resource in Twilio. Then you can create a new "participant" for each MMS user and add them to the conversation. From there, every user attached to the conversation should receive all messages. In my case, it seems to work.

https://www.twilio.com/docs/conversations
